File: packages/js-dpp/lib/errors/DataContractNotPresentError.js
const DPPError = require('./DPPError');

class DataContractNotPresentError extends DPPError {
  /**
   * @param {Identifier} dataContractId
   */
  constructor(dataContractId) {
    super('Data Contract is not present');

    this.dataContractId = dataContractId;
  }

  /**
   * Get Data Contract ID
   *
   * @return {Identifier}
   */
  getDataContractId() {
    return this.dataContractId;
  }
}

module.exports = DataContractNotPresentError;

File: packages/rs-dpp/src/data_contract/errors/data_contract_not_present_error.rs
use platform_value::Identifier;
use thiserror::Error;

use crate::ProtocolError;

// @append_only
#[derive(Error, Debug, Clone, PartialEq, Eq)]
#[error("Data Contract is not present")]
pub struct DataContractNotPresentError {
    data_contract_id: Identifier,
}

impl DataContractNotPresentError {
    pub fn new(data_contract_id: Identifier) -> Self {
        Self { data_contract_id }
    }

    pub fn data_contract_id(&self) -> Identifier {
        self.data_contract_id
    }
}

impl From<DataContractNotPresentError> for ProtocolError {
    fn from(err: DataContractNotPresentError) -> Self {
        Self::DataContractNotPresentError(err)
    }
}
